d2/src/test/java/com/linkedin/d2/balancer/servers/ZookeeperConnectionManagerTest.java

```diff
@@ -352,7 +353,7 @@ public void testMarkUpDuringSessionExpiration()
     shutdownManager(manager);
   }
 
-  @Test(invocationCount = 10, timeOut = 10000)
+  @Test(invocationCount = 10, timeOut = 10000, retryAnalyzer = SingleRetry.class) // TODO: temporary retry to alleviate CI failures, should be investigated
```

Comment:
It is not necessary but I think 3 times will be better, sort of industry standard?

Reply:
I can create another implementation for 3 retries. I think I will add this as part of the dev guide:
- If test is flaky in CI, add single retry.
- If still flaky, use three retries.
- If still flaky, add to `ci-flaky` group.

Or do you think this is excessive?

Reply:
Second Junchuan, 3 times retry is kind of industry standard. Any reason to set it for 10?

Reply:
Did you check the tests in ci-flaky group with 3 retires?

Reply:
@nickibi The retry here (`SingleRetry`) allows for one retry. The `invocationCount = 10` expression means that the test will run 10 times and must pass all 10 times.

@karthikbalasub Nope, I excluded tests that failed even with a retry. I'll try another iteration that replaces `ci-flaky` with 3 retries. Then we'll see which ones really need to be excluded.